### PR TITLE
Enable app ingress TLS by default and fix Route CRD

### PR DESCRIPTION
**Pull request by XanderStrike**
* This bump of CF K8s Networking includes two changes.
  1. Removes the Status section from the Route CRD
      [#172569406](https://www.pivotaltracker.com/story/show/172569406).
  2. Changes which gateway is associated with the virtual services for
      app routes. This is part of the work needed to have app ingress routing
      support TLS by default.

* This also includes the changes to cf-for-k8s to enable app TLS by default
  1. New requirement for distinct app and system domains is documented
      in values files
  2. Fields added for new cert to values files
  3. generate-values now creates a 2nd cert for app ingress
      automatically

fixes cloudfoundry/cf-for-k8s#46

**Acceptance Steps**

```
GIVEN a fresh kubernetes cluster
WHEN I install cf-for-k8s
AND I deploy an application
THEN I can reach my app via https
```

THE SQUAD: @jenspinney @ndhanushkodi @tcdowney @christianang @kauana @jrussett

### CODE REVIEW

File: config/_ytt_lib/github.com/cloudfoundry/cf-k8s-networking/config/crd/networking.cloudfoundry.org_routes.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (jamespollard8) [inline config/_ytt_lib/github.com/cloudfoundry/cf-k8s-networking/config/crd/networking.cloudfoundry.org_routes.yaml:1]: nit: is this blank line intentional?

Comment (tcdowney) [reply config/_ytt_lib/github.com/cloudfoundry/cf-k8s-networking/config/crd/networking.cloudfoundry.org_routes.yaml:1]: Yeah, Kubebuilder adds that blank line to its generated CRDs for some reason (maybe not intentional on their end). See the example `CronJob` CRD:

https://github.com/kubernetes-sigs/kubebuilder/blob/d6e2b995831ce26228f2303e9d1b3a55acd3d225/docs/book/src/cronjob-tutorial/testdata/project/config/crd/bases/batch.tutorial.kubebuilder.io_cronjobs.yaml#L1

We had manually copy pasted it to our top level `config` directly previously I think which is why the blank line shows up in the diff here. This time I copied the generated file entirely.

**Comment by jenspinney** [comment]
> Hello friend, it looks like your pull request has failed one or more of our checks. Please take a look! 👀

The tests failed because **we've added a new restriction that the system domain and apps domain be different**. The cf-for-k8s pipeline assumed they were the same. I've pushed a commit to this branch that I hope will fix it, but this may take some trial-and-error to get right.

**Comment by jenspinney** [comment]
Actually, the smoke test run here won't reconfigure the pipeline, so I expect this still to fail. 🤔

**Comment by jenspinney** [comment]
I think we'll have to do this PR in two steps:

1. A PR to change the generate-values script & cf-for-k8s CI so that the system domain and apps domain are different. After merging, Rel-int reconfigures their CI with the pipeline changes.
1. This PR, with the changes to the Route CRD and the functionality needed to have app ingress TLS working by default.

**Comment by jenspinney** [comment]
After discussing with folks on Rel-Int, closing this PR in favor of just PRing the Route CRD change as a standalone thing and investigating whether we can continue to use a single domain for apps and system components in the rel-int CI.

**Pull request closed without merging**